diff --git a/MagickCore/delegate.c b/MagickCore/delegate.c
index 4214c9e..a507b90 100644
--- a/MagickCore/delegate.c
+++ b/MagickCore/delegate.c
@@ -84,7 +84,7 @@
 */
 #if defined(__APPLE__)
   #include "TargetConditionals.h"
-  #if TARGET_OS_IOS || TARGET_OS_WATCH || TARGET_OS_TV
+  #if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
     #define system(s) ((s)==NULL ? 0 : -1)
   #endif // end iOS
 #elif defined(__ANDROID__)
diff --git a/MagickCore/random.c b/MagickCore/random.c
index a02aebe..63145ce 100644
--- a/MagickCore/random.c
+++ b/MagickCore/random.c
@@ -64,6 +64,9 @@
 #include "MagickCore/timer-private.h"
 #include "MagickCore/utility.h"
 #include "MagickCore/utility-private.h"
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#endif
 #if defined(MAGICKCORE_HAVE_GETENTROPY)
 #include <sys/random.h>
 #endif
@@ -118,7 +121,7 @@ struct _RandomInfo
 /*
   External declarations.
 */
-#if defined(__APPLE__) && !defined(TARGET_OS_IPHONE)
+#if defined(__APPLE__) && (!defined(TARGET_OS_IPHONE) || !TARGET_OS_IPHONE)
 #include <crt_externs.h>
 #define environ (*_NSGetEnviron())
 #endif
