--- temp_osxcross/tools/tools.sh	2026-03-17 06:23:46.724429343 +0000
+++ tools/tools.sh	2026-03-16 01:41:37.042355648 +0000
@@ -183,7 +183,7 @@
   local res=$(x86_64-apple-$OSXCROSS_TARGET-lipo -detailed_info \
               "$1/usr/lib/crt1.o")
 
-  if [[ $res == *architecture\ ppc* ]] &&
+  if [[ $res == *architecture\ ppc* ]] ||
      [[ $res == *architecture\ ppc64* ]]; then
     echo 1
   else
