--- a/build_gcc.sh	2026-03-17 06:23:45.093987918 +0000
+++ b/build_gcc.sh	2026-03-16 01:41:36.966262542 +0000
@@ -53,10 +53,9 @@
     echo "The SDK you are using does not support PowerPC" 1>&2
     exit 1
   fi
-  GCC_ARCH=powerpc64
-  GCC_ARCH_SHORT=oppc64
-  GCC_ARCH32=powerpc
-  GCC_ARCH_SHORT32=oppc32
+  GCC_ARCH=powerpc
+  GCC_ARCH_SHORT=oppc32
+  GCC_ARCH_SHORT32=oppc32
   TARGETARCHS=ppc
   if [ -z "$APPLE_GCC" ]; then
     export DISABLE_ANNOYING_LD64_ASSERTION=1
@@ -76,7 +75,11 @@
 
 if [ -n "$APPLE_GCC" ]; then
   # work around broken multilib support
-  ARCHS_TO_BUILD+=" $GCC_ARCH32"
+  # Forced to 32-bit PPC only
+  ARCHS_TO_BUILD="powerpc"
+  GCC_ARCH_SHORT="oppc32"
+  GCC_ARCH_SHORT32="oppc32"
+  EXTRA_CONFFLAGS+="--disable-multilib "
   EXTRA_CONFFLAGS+="--with-gxx-include-dir=/usr/include/c++/4.0.0 " ## TODO: version
 
   # fix -lstdc++ link
